projects/aca-content/src/lib/components/header-actions/header-actions.component.html

```diff
@@ -0,0 +1,48 @@
+<div>
+  <div class="adf-toolbar--spacer adf-toolbar-divider"></div>
+  <span class="action-bar">
+    <button class="aca-mat-button aca-create-button" mat-stroked-button data-automation-id="create-button"
+      *ngIf=" isPersonalFilesRoute() || isFavoriteLibrariesRoute() || isLibrariesrRoute() "
+      [matMenuTriggerFor]="createMenu">
+      {{ 'APP.HEADER.BUTTONS.CREATE' | translate }}
+    </button>
+    <mat-menu #createMenu="matMenu" role="menu" class="app-create-menu__root-menu app-create-menu__sub-menu"
+      [overlapTrigger]="false" yPosition="below">
+      <div *ngFor="let action of createActions; trackBy: trackByActionId">
+        <app-toolbar-menu-item [actionRef]="action"></app-toolbar-menu-item>
+      </div>
+    </mat-menu>
+
+    <button class="aca-mat-button aca-upload-button" mat-stroked-button data-automation-id="upload-button"
+      *ngIf="isPersonalFilesRoute()" [matMenuTriggerFor]="uploadMenu">
```

Comment:
please rename to "canShowUploadMenu" and evaluate all conditions there... don't bind UI to the knowledge of "personal files route" as there can be other conditions and routes in the future